### PR TITLE
gh-86618: Improve colorsys.rgb_to_hls code

### PR DESCRIPTION
**Pull request by jjerphan**
Cache repeated sum and difference to make code slightly faster and easier to read.

* Issue: gh-86618

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@jjerphan

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by jstasiak — CHANGES_REQUESTED** [review]
I hate to be that guy, but since this is about optimization I think some before/after benchmarks need to be included together with the test data and some details of the test environment.

**Comment by Fidget-Spinner** [comment]
> I hate to be that guy, but since this is about optimization I think some before/after benchmarks need to be included together with the test data and some details of the test environment.

@jstasiak I'm not sure if I'm benchmarking wrongly, but I don't see any performance improvements either:
```python
import timeit
timeit.timeit(stmt="rgb_to_hls(255, 255, 255)", setup="from colorsys import rgb_to_hls")
```
On main branch:
`2.8296601999999993`

On this PR's branch:
`2.8520722999999997`

Basically no significant change.

**Comment by jjerphan** [comment]
Using this simple — yet more comprehensive — script which uses the test data, as recommended by @jstasiak :

```python
# /usr/bin/env python

import timeit
import statistics
from Lib.test.test_colorsys import ColorsysTest


if __name__ == '__main__':
    number = 100
    repeat = 100
    results = timeit.repeat('ColorsysTest().test_hls_roundtrip()',
                    setup='from Lib.test.test_colorsys import ColorsysTest',
                    number=number,
                    repeat=repeat)

    running_times = list(sorted(results))

    mean = statistics.mean(running_times)
    std = statistics.stdev(running_times)
    print(f"Running time: {mean} ± {std} ms.  "
    f"(number, repeat) = ({number}, {repeat})")
```

I get on:
 - on a26215db11cfcf7b5f55cab9e91396761a0e0bcf (i.e. `HEAD~1`)
```
Running time: 0.13036188921425493 ± 0.003873555156404328 ms.  (number, repeat) = (100, 100)
```
 - on becb854c3e9637956f286b305151f7007d0ced13 (i.e. `HEAD`)
```
Running time: 0.12722085006709677 ± 0.0033936437887972116 ms.  (number, repeat) = (100, 100)
```

Testing environment:
```bash
$ uname -a ; gcc -v
Linux torus 5.8.15-201.fc32.x86_64 #1 SMP Thu Oct 15 15:56:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
Using built-in specs.
COLLECT_GCC=/usr/bin/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/10/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl --enable-offload-targets=nvptx-none --without-cuda-driver --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.2.1 20201016 (Red Hat 10.2.1-6) (GCC) 
```

**Comment by jjerphan** [comment]
@jstasiak: I guess it would be better to close this PR, open a bpo-issue and attach a new PR to it. What do you think?

**Comment by Fidget-Spinner** [comment]
> @jstasiak: I guess it would be better to close this PR, open a bpo-issue and attach a new PR to it. What do you think?

You can open the bpo issue then prepend the issue to the PR title, it will then auto link :).

**Comment by jjerphan** [comment]
Done.

File: Lib/colorsys.py [hunk 1]
```diff
@@ -75,17 +75,18 @@ def yiq_to_rgb(y, i, q):
...
         return 0.0, l, 0.0
     if l <= 0.5:
-        s = (maxc-minc) / (maxc+minc)
+        s = rangec / sumc
     else:
-        s = (maxc-minc) / (2.0-maxc-minc)
-    rc = (maxc-r) / (maxc-minc)
-    gc = (maxc-g) / (maxc-minc)
-    bc = (maxc-b) / (maxc-minc)
+        s = rangec / (2.0-sumc)
```
Comment (terryjreedy) [inline Lib/colorsys.py:86]: Verifying: (2-maxc-minc) = (2-(maxc+minc)) = (2-sumc)
This make the parallel with the 'if' denominator more evident.

**Review by terryjreedy — APPROVED** [review]
I believe this module is one of those proposed for removal.  But that proposal was rejected for 3.9 and the module has not yet been deprecated.

Addition and subtraction are so fast that I am not surprised that a speedup in hard to detect.  But the change makes the code easier to read.  Since the PR is made and the change is easy enough to verify as correct and this function is thoroughly tested, I think it worth making.

While this change is conceptually trivial, the code change itself is not, for the purpose skipping the CLA and bpo issue.  I will, however, skip a news item.

There is no report available for the Pipelines failure.  Given the nature of the change and its passing on multiple other tests, I am going to assume that the Pipelines failure is unrelated.

**Pull request merged**